sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java

```diff
@@ -131,10 +128,22 @@
         Response<SentimentResponse> response) {
         final SentimentResponse sentimentResponse = response.getValue();
         final List<AnalyzeSentimentResult> analyzeSentimentResults = new ArrayList<>();
-        for (DocumentSentiment documentSentiment : sentimentResponse.getDocuments()) {
-            analyzeSentimentResults.add(convertToAnalyzeSentimentResult(documentSentiment));
+        final List<DocumentSentiment> documentSentiments = sentimentResponse.getDocuments();
+        for (DocumentSentiment documentSentiment : documentSentiments) {
+            analyzeSentimentResults.add(convertToAnalyzeSentimentResult(documentSentiment, documentSentiments));
         }
         for (DocumentError documentError : sentimentResponse.getErrors()) {
+            /*
+             *  TODO: Remove this after service update to throw exception.
```

Comment:
this should be fixes by now, no? are u planning on updating our code to delete this condition in a future PR?

Reply:
this will be take care in another PR. I will remove them one time at all.

Reply:
#10962